tests/unit/test_auth_middleware.py

```
@@ -0,0 +1,330 @@
+    def test_bypass_mode_no_token_needed(self, client: TestClient, mock_settings):
+        """Bypass mode should not require token."""
+        mock_settings.cloudflare_enabled = False
+
+        response = client.get("/api/v1/user/me")
+        assert response.status_code == 200
+
+
+class TestJWKSCaching:
+    """Tests for JWKS caching behavior."""
+
+    def test_jwks_cache_reused(self, client: TestClient, mock_settings):
```

Ruff (0.14.8)

Unused method argument: `mock_settings`

(ARG002)